universal-application-tool-0.0.1/test/services/PathTest.java

```diff
@@ -137,4 +137,14 @@ public void pathBuilder() {
     path = path.toBuilder().setPath("something.new").build();
     assertThat(path.path()).isEqualTo("something.new");
   }
+
+  @Test
+  public void pathJoin() {
+    Path path = Path.empty();
+    path = path.join("applicant.my.path");
```

Comment:
I think this test case reveals a bug in `Path#join`. The `"applicant.my.path"` segment is added as a single segment to the path.

So 
```
Path path = Path.create("path.start");
Path anotherPath = path.join("a.long.segment");
assertThat(anotherPath.parentPath()).isEqualTo(path);
```
But I think we'd really want `anotherPath.parentPath()` to be `Path.create(path.start.a.long)`

Reply:
A fix would be to update the word `segment` to `path` in the docstring and implementation, and then fix `join` so that it properly joins paths.

You're welcome to open a separate PR to bugfix that, or I can make it an issue for another time.

Reply:
I believe I've correctly searched for usages, let me know what you think of the newest commit

Reply:
This bug was fixed in #680